src/rgw/rgw_asio_frontend.cc

```diff
@@ -34,19 +36,23 @@ template <typename Stream>
 class StreamIO : public rgw::asio::ClientIO {
   CephContext* const cct;
   Stream& stream;
-  beast::flat_buffer& buffer;
+  boost::asio::yield_context yield;
+  boost::beast::flat_buffer& buffer;
  public:
   StreamIO(CephContext *cct, Stream& stream, rgw::asio::parser_type& parser,
-           beast::flat_buffer& buffer, bool is_ssl,
+           boost::asio::yield_context yield,
+           boost::beast::flat_buffer& buffer, bool is_ssl,
            const tcp::endpoint& local_endpoint,
            const tcp::endpoint& remote_endpoint)
       : ClientIO(parser, is_ssl, local_endpoint, remote_endpoint),
-        cct(cct), stream(stream), buffer(buffer)
+        cct(cct), stream(stream),
+        yield(yield), buffer(buffer)
   {}
 
   size_t write_data(const char* buf, size_t len) override {
     boost::system::error_code ec;
-    auto bytes = boost::asio::write(stream, boost::asio::buffer(buf, len), ec);
+    auto bytes = boost::asio::async_write(stream, boost::asio::buffer(buf, len),
+                                          yield[ec]);
```

Comment:
I don't think we're ready for this `rgw: beast frontend uses yield_context to read/write body` commit until we finish most of the log refactoring work - yielding here would mean that log output is split between threads

doing the async scheduling is fine, because process_request() has already been refactored and all of rgw_process_authenticated() should stay on the same thread

Reply:
dropping, I thought I had caught that. Thanks